src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/WorldAnchor/WorldAnchorSpatialLocalizer.cs

```diff
@@ -63,47 +63,62 @@ public override bool TryCreateLocalizationSession(IPeerConnection peerConnection
             {
                 this.localizer = localizer;
                 this.settings = settings;
                 this.peerConnection = peerConnection;
             }
 
-            public async Task<ISpatialCoordinate> LocalizeAsync(CancellationToken cancellationToken)
+            /// <inheritdoc />
+            public override async Task<ISpatialCoordinate> LocalizeAsync(CancellationToken cancellationToken)
             {
-                WorldAnchorCoordinateService coordinateService = await localizer.coordinateServiceTask.Unless(cancellationToken);
-                if (cancellationToken.IsCancellationRequested)
+                if (!defaultCTS.Token.CanBeCanceled)
                 {
+                    Debug.LogError("Session is invalid. No localization performed.");
                     return null;
                 }
 
-                if (settings.Mode == WorldAnchorLocalizationMode.LocateExistingAnchor)
+                ISpatialCoordinate spatialCoordinate = null;
+                using (var cancellableCTS = CancellationTokenSource.CreateLinkedTokenSource(defaultCTS.Token, cancellationToken))
                 {
-                    if (coordinateService.TryGetKnownCoordinate(settings.AnchorId, out ISpatialCoordinate coordinate))
+                    WorldAnchorCoordinateService coordinateService = await localizer.coordinateServiceTask.Unless(cancellationToken);
+                    if (cancellationToken.IsCancellationRequested)
                     {
-                        return coordinate;
+                        return null;
+                    }
+
+                    if (settings.Mode == WorldAnchorLocalizationMode.LocateExistingAnchor)
+                    {
+                        if (coordinateService.TryGetKnownCoordinate(settings.AnchorId, out ISpatialCoordinate coordinate))
+                        {
+                            return coordinate;
+                        }
+                        else
+                        {
+                            return null;
+                        }
                     }
                     else
                     {
-                        return null;
+                        spatialCoordinate = await coordinateService.CreateCoordinateAsync(settings.AnchorId, settings.AnchorPosition, settings.AnchorRotation, cancellationToken);
                     }
                 }
-                else
-                {
-                    return await coordinateService.CreateCoordinateAsync(settings.AnchorId, settings.AnchorPosition, settings.AnchorRotation, cancellationToken);
-                }
+
+                return await Task.FromResult(spatialCoordinate);
             }
 
-            public void OnDataReceived(BinaryReader reader)
+            /// <inheritdoc />
+            public override void OnDataReceived(BinaryReader reader)
             {
             }
```

Comment:
If this is now virtual and we don't need to explicitly not have the base method happen, we can just get rid of this empty method now